Makefile

```diff
@@ -28,163 +27,157 @@ export IRONIC_INSPECTOR_ENDPOINT=http://localhost:5050/v1/
+	./hack/gofmt.sh
+
+vet: ## Run go vet against code
+	go vet ./...
 
 .PHONY: lint
 lint: golint-binary ## Run golint
 	find $(CODE_DIRS) -type f -name \*.go  |grep -v zz_ | xargs -L1 golint -set_exit_status
 
+.PHONY: golint-binary
+golint-binary:
+	which golint 2>&1 >/dev/null || $(MAKE) $(GOPATH)/bin/golint
+$(GOPATH)/bin/golint:
+	go get -u golang.org/x/lint/golint
+
+generate: controller-gen ## Update generated source code
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./api/..."
+
 .PHONY: generate-check
-generate-check:
+generate-check: ## Verify that generated files are checked in
 	./hack/generate.sh
 
 .PHONY: generate-check-local
-generate-check-local:
+generate-check-local: ## Verify that generated files are checked in without using a container
 	IS_CONTAINER=local ./hack/generate.sh
 
 .PHONY: sec
 sec: $(GOPATH)/bin/gosec ## Run gosec
-	gosec -severity medium --confidence medium -quiet $(PACKAGES)
+	gosec -severity medium --confidence medium -quiet ./...
 
 $(GOPATH)/bin/gosec:
 	go get -u github.com/securego/gosec/cmd/gosec
 
-.PHONY: golint-binary
-golint-binary:
-	which golint 2>&1 >/dev/null || $(MAKE) $(GOPATH)/bin/golint
-$(GOPATH)/bin/golint:
-	go get -u golang.org/x/lint/golint
-
-.PHONY: fmt
-fmt: ## Run gofmt and write changes to each file
-	gofmt -l -w $(CODE_DIRS)
-
-.PHONY: fmt-check
-fmt-check: ## Run gofmt and report an error if any changes are made
-	./hack/gofmt.sh
-
-.PHONY: docs
-docs: $(patsubst %.dot,%.png,$(wildcard docs/*.dot))
-
-%.png: %.dot
-	dot -Tpng $< >$@
-
-.PHONY: e2e-local
-e2e-local:
-	operator-sdk test local ./test/e2e \
-		--namespace $(TEST_NAMESPACE) \
-		--up-local $(SETUP) \
-		$(DEBUG) --go-test-flags "$(GO_TEST_FLAGS)"
-
-.PHONY: run
-run: ## Run the operator outside of a cluster in development mode
-	operator-sdk run --local \
-		--go-ldflags=$(LDFLAGS) \
-		--watch-namespace=$(RUN_NAMESPACE) \
-		--operator-flags="-dev"
-
-.PHONY: demo
-demo: ## Run the operator outside of a cluster using the demo driver
-	operator-sdk run --local \
-		--go-ldflags=$(LDFLAGS) \
-		--watch-namespace=$(RUN_NAMESPACE) \
-		--operator-flags="-dev -demo-mode"
+.PHONY: docker-build
+docker-build: test ## Build the docker image
+	docker build . -t ${IMG}
 
+# Compatibility alias from older version of this file
 .PHONY: docker
-docker: docker-operator docker-sdk docker-golint ## Build docker images
+docker: docker-build
+
+.PHONY: docker-push
+docker-push: ## Push the docker image
+	docker push ${IMG}
+
+# find or download controller-gen
+# download controller-gen if necessary
+controller-gen:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION) ;\
+	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
+	}
+CONTROLLER_GEN=$(GOBIN)/controller-gen
+else
+CONTROLLER_GEN=$(shell which controller-gen)
+endif
```

Comment:
you could put this at the top
```
ifeq (, $(shell which controller-gen))
CONTROLLER_GEN=$(GOBIN)/controller-gen
else
CONTROLLER_GEN=$(shell which controller-gen)
endif
```
then
```
$(CONTROLLER_GEN):
   #install stuff
```
just to avoid the phony target and repeated installs

Reply:
actually isn't it better to vendor? if the generated code varies across versions that's going to be a pain

Reply:
The install logic uses the specific version we need. It's based on what kubebuilder generates. We could enhance it to install into a local directory, like with kustomize, in another PR if we think that's important.